src/core/model/types/double_type.h

```diff
@@ -12,7 +13,7 @@ class DoubleType final : public NumericType<Double> {
         Double l_val = GetValue(l);
         Double r_val = GetValue(r);
 
-        if (std::abs(l_val - r_val) < std::numeric_limits<Double>::epsilon()) {
+        if (util::AlmostEqualUlps(l_val, r_val, 10)) {
```

Comment:
Maybe 10 as a number of ulps works fine in some cases, but I'm not sure that we should use it for every double comparison at least without alternative method which allows to specify the precision. The desired precision actually depends on the nature of doubles we're comparing and hence on the task we're solving, for some algorithm 10 might be either too large or too small.

Reply:
Also, please, stop using [magic number anti-pattern](https://en.wikipedia.org/wiki/Magic_number_(programming)) and replace all non-obvious values (that is all numbers besides 0 and 1 in most cases) with named constants. This regards all the code, including tests.